### ISSUE TITLE
Target count off by 1 (Based on damage)

### ISSUE THREAD

**Issue by behub-au** [Hekili/hekili#74]
Noticed this in raid the other day so went out to test it. Only noticed it recently after one of the updates.

AoE Count on right icon
![image](https://user-images.githubusercontent.com/870239/53758195-73204780-3f11-11e9-939b-a96891a93405.png)

Nameplate detection seems to be ok
![image](https://user-images.githubusercontent.com/870239/53758229-93500680-3f11-11e9-97de-b0c628c06f73.png)

**Comment by Hekili** [comment]
So, there's probably some further refinement that needs to be sorted, but target detection is fairly complicated for a number of reasons.

![image](https://user-images.githubusercontent.com/9438134/53759973-93a7bd80-3e8f-11e9-8962-c6e7e44dc057.png)

Nameplate Detection is great for melee, since you can reliably tell what is within a specific radius around you.

Detect Enemies by Damage is a bit more complex, and refers to *direct damage*.  I.e., if you hit something recently, you want to count it as a target.

Detect Enemies by Damage over Time is another layer of complexity, because let's say you're a Destruction Warlock and you just popped Cataclysm(?) and put Immolate on 8 targets.  You *probably* want to consider all of those valid targets, but what about 5 seconds later?  If they're still clustered, then maybe you do want to count all 8 and drop a Rain of Fire on them.  But what if they all ran away in different directions and are nowhere close to each other?  Yes, they're still valid targets, but maybe they're not clustered in a way that's useful to you any longer.

Count targets and deciding when they're valid for multidotting, etc, is something I'll continue to refine.  Bringing this back to your case with the Balance Druid above, it's possible that you don't count targets based on Damage over Time -- check your settings and make a judgment call.

I've also posted -beta11 that is more friendly when counting dummies, because dummies are special cases.  You can do "overkill" damage to them, which the addon now uses to decide to stop counting a target, even though the dummy doesn't actually die.

tl;dr:  Try beta11 and check out the target handling options for each of your specs and let me know if you still find things inconsistent.

**Comment by behub-au** [comment]
I notced the DoT setting after i posted the issue. I was curious more because even Lunar Strike hitting 3 targets didn't change that number during raid, so it felt like something had changed recently.

When i enable the detect enemies by DoT, even with 7 dots out on 7 targets it still says only 6 targets. I guess that is where I am unsure on the counter.

**Comment by Hekili** [comment]
Well, try the update and see how that goes. And make sure to report on what
version you're using at the time.

**Comment by behub-au** [comment]
I'll update when i get home from work. On beta 10 at the moment but Twitch/curse takes hours to hit whatever CDN I am using.

**Comment by behub-au** [comment]
Running version 12 now and it seems to work better, whatever changes you did.
